msteams-platform/tabs/how-to/personal-tab-navigation.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Personal tab app navigation
+description: Learn about the options for navigation within a personal tab app using the TeamsJS library.
+ms.date: 10/26/2022
+ms.topic: conceptual
+ms.custom: m365apps
+ms.localizationpriority: medium
+---
+
+# Navigate within a personal tab app
+
+The TeamsJS library provides support for navigation within a personal app. This article discusses the options available, which include types of navigation such as between tabs within the app or through use of Teams UI components and the back button.
+
+[!INCLUDE [sdk-include](~/includes/sdk-include.md)]
+
+Personal tabs provide a great way to enhance the Microsoft Teams experience. You can provide the user access to your web application right within Teams using personal tabs, without having to sign in again. For more information about personal tabs and how you can extend them across Microsoft 365 products, see [Build tabs for Teams](~/tabs/what-are-tabs.md) and [Extend a Teams personal tab across Microsoft 365](~/m365-apps/extend-m365-teams-personal-tab.md).
+
+## Navigate between tabs
+
+The [pages](/javascript/api/@microsoft/teams-js/pages?view=msteams-client-js-latest&preserve-view=true) capability of the TeamsJS library provides support for navigation between tabs within a personal app. Specifically, the [pages.currentApp](/javascript/api/@microsoft/teams-js/pages.currentapp?view=msteams-client-js-latest&preserve-view=true) namespace offers a function `navigateTo(NavigateWithinAppParams)` to allow navigation to a specific tab within the current app and a function `navigateToDefaultPage()` to navigate to the first tab defined in the app's manifest.
+
+The following code illustrates how to navigate to a specific page:
+
+```js
+if (pages.currentApp.isSupported()) {
```

Comment:
So I was checking and saw that teams Hub has yet to support the pages.currentApp namespace even though that is released in the latest version of the TeamsJS SDK. I talked to an engineering manager who said that it would be best to make two sections(one for hubs that do support and one for those who do not). This is because Outlook and Office are supposed to have support for this very soon.

Reply:
@vikramtha
Do you require them to be appearing in 2 different tabs? Hubs that do support and that do not.

Reply:
@vikramtha @v-amprasad - I think this might be worth a conversation on later today  at our sync @vikramtha. Given we show doing the isSupported check for the varying support in the different products, I would argue against breaking it out. Setting a practice of breaking out support for each teamsjs call is just going to be a mess and hard to keep track of. I'd rather see it as part of the API ref or as a single table somewhere if anything. If not having the Teams support yet is an issue, I'd suggest holding off on the article or potentially publishing this info in a different way (part of our Tab extension articlie) until the full support is there.